Sources/SwiftGRPCNIO/ClientErrorDelegate.swift

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+
+public protocol ClientErrorDelegate: class {
+  /// Called when the client catches an error.
+  ///
+  /// - Parameters:
+  ///   - error: The error which was caught.
+  ///   - file: The file where the error was raised.
+  ///   - line: The line within the file where the error was raised.
+  func didCatchError(_ error: Error, file: StaticString, line: Int)
+}
+
+public class LoggingClientErrorDelegate: ClientErrorDelegate {
```

Comment:
Should we offer a shared instance of this? (Related to the weak reference comment above.)

Reply:
Sure, we can do that.

Reply:
Let's do that and make `init` private :-)